extension/services/soundcloud/soundcloud.js

```diff
@@ -0,0 +1,74 @@
+import * as music from "../../intents/music/music.js";
+import * as serviceList from "../../background/serviceList.js";
+import * as content from "../../background/content.js";
+import { shouldDisplayWarning } from "../../limiter.js";
+
+class Soundcloud extends serviceList.Service {
```

Comment:
It's possible we should be making an abstract base class that both Spotify and Soundcloud (and future services) inherit from.

Reply:
I have added an abstract class of MusicService but I don't know if this is what you meant. If not, can you give some pointer how can I do that?

Reply:
Ah, cool, yes, what you did is what I meant